x-pack/plugins/security_solution/server/endpoint/routes/metadata/enrichment.test.ts

```diff
@@ -0,0 +1,220 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { HostStatus, MetadataQueryStrategyVersions } from '../../../../common/endpoint/types';
+import { createMockMetadataRequestContext } from '../../mocks';
+import { EndpointDocGenerator } from '../../../../common/endpoint/generate_data';
+import { enrichHostMetadata, MetadataRequestContext } from './handlers';
+
+describe('test document enrichment', () => {
+  let metaReqCtx: jest.Mocked<MetadataRequestContext>;
+  const docGen = new EndpointDocGenerator();
+
+  beforeEach(() => {
+    metaReqCtx = createMockMetadataRequestContext();
+  });
+
+  // verify query version passed through
+  describe('metadata query strategy enrichment', () => {
+    it('should match v1 strategy when directed', async () => {
+      const enrichedHostList = await enrichHostMetadata(
+        docGen.generateHostMetadata(),
+        metaReqCtx,
+        MetadataQueryStrategyVersions.VERSION_1
+      );
+      expect(enrichedHostList.query_strategy_version).toEqual(
+        MetadataQueryStrategyVersions.VERSION_1
+      );
+    });
+    it('should match v2 strategy when directed', async () => {
+      const enrichedHostList = await enrichHostMetadata(
+        docGen.generateHostMetadata(),
+        metaReqCtx,
+        MetadataQueryStrategyVersions.VERSION_2
+      );
+      expect(enrichedHostList.query_strategy_version).toEqual(
+        MetadataQueryStrategyVersions.VERSION_2
+      );
+    });
+  });
+
+  describe('host status enrichment', () => {
+    let statusFn: jest.Mock;
+
+    beforeEach(() => {
+      statusFn = jest.fn();
+      (metaReqCtx.endpointAppContextService.getAgentService as jest.Mock).mockImplementation(() => {
+        return {
+          getAgentStatusById: statusFn,
+        };
+      });
```

Comment:
Do we have to do the casting, this is a suggested alternative. Here we set the mocks, this is just an example. 
```
       endpointAppContextService.start({
         ...createMockEndpointAppContextServiceStartContract(),
        ...{ agentPolicyService: mockAgentPolicyService, agentService: mockAgentService },
       });

```

Reply:
the casting I had to do to get access to the jest properties. Since through the few levels of mocked interfaces, it's returning real types, though mocked objects.

But this was enough to allow me to just mock the things I need, e.g. on line 178:

```
      agentPolicyMock.mockImplementation(() => {
        return {
          id: policyID,
          revision: policyRev,
        };
      });
```

and differently on 198

```
      agentPolicyMock.mockImplementation(() => {
        return {
          package_policies: [
            {
              package: { name: 'endpoint' },
              id: policyID,
              revision: policyRev,
            },
          ],
        };
      });
```